# Activity 8

This activity tests your skills
in formulating an engineering problem as solving a system of linear algebraic equations,
as well as analyzing results thereof.

## Scenario

The lateral view of an [inclined strut](./act-08.png)
is designed as a ten-member truss.

You are tasked to analyze a scenario wherein
loads are approximated as concentrated forces 
acting on the non-support joints.
Members $AC$, $CE$, and $EG$ are collinear,
and perpendicular to the lines of action of loads $p_{1}$, $p_{2}$, and $p_{3}$.
Members $BD$, $DF$, and $FG$ are collinear,
and perpendicular
not just to the lines of action of loads $p_{4}$ and $p_{5}$,
but also to the supporting surface.

Let the dimensions, material makeup, and weight of the members be negligible.
Assume that the joints do not resist moments.
Determine the member forces developed
under such a conservative idealization of the strut.

## Modelling

Cast the original task as solving a system of linear algebraic equations
for the forces in $AC$, $BC$, $BD$, $CD$, $CE$, $DE$, $DF$, $EF$, $EG$, and $FG$.

Arrange the equations in the matrix-vector form
$\boldsymbol{K} \boldsymbol{\phi} = \boldsymbol{\delta}$,
where
$\boldsymbol{\phi}$ collects the member forces in the order presented above.
For uniformity, assume that tensile forces are positive.

## Core code

In [1]:
import math as mt

import numpy as np

Define a Python function
`get_coeffs_rhs()`
that takes four positional arguments
(namely, `d`, `h`, `theta`, `p1`, `p2`, `p3`, `p4`, `p5`,
in that order)
and returns two NumPy arrays
(namely,
`coeff_matrix`, `rhs_vector`
in that order).
The roles of these inputs and outputs should be evident.

Derived Matrix: [Matrix1](.1_a8.jpg)
[Matrix2](.2_a8.jpg)

MY MODEL SOLUTION WAS DERIVED BY NOT USING THE ANGLE THETA(I assigned h as the y-axis and the base of the strut as x-axis)
basically tilting the angle of the strut

In [2]:
def get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang): #phi as the angle I use instead of theta 
    
    BC_x_vec = d/(mt.sqrt(d**2 + (2*h/3)**2))
    BC_y_vec = 2*h/(3*mt.sqrt(d**2 + (2*h/3)**2))
    
    DE_x_vec = d/(mt.sqrt(d**2 + (h/3)**2))
    DE_y_vec = h/(3*mt.sqrt(d**2 + (h/3)**2))
    
    coeff_matrix = np.array ([
        [0, 0, 0, 0, 0, 0, 0, 0, -mt.cos(phi_ang), -1],
        [0, 0, 0, 0, 0, 0, 0, 0, mt.sin(phi_ang), 0],
        [0, 0, 0, 0,-mt.cos(phi_ang), -DE_x_vec, 0, 0, -mt.cos(phi_ang), 0],
        [0, 0, 0, 0, mt.sin(phi_ang), -DE_y_vec, 0, -1, mt.sin(phi_ang), 0],
        [0, 0, 0, 0, 0, 0, -1, 0, 0, -1],
        [0, 0, 0, 0, 0, 0, 0, -1, 0, 0],
        [0, 0, -1, 0, 0,-DE_x_vec, -1, 0, 0, 0],
        [0, 0, 0, 1, 0, -DE_y_vec, 0, 0, 0, 0],
        [-mt.cos(phi_ang), -BC_x_vec, 0, 0, -mt.cos(phi_ang), 0, 0, 0, 0, 0],
        [mt.sin(phi_ang), -BC_y_vec, 0, 1, mt.sin(phi_ang), 0, 0, 0, 0, 0],
    ])
    return coeff_matrix 

def delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang): 
    rhs_vector = np.array([p3*mt.sin(phi_ang), p3*mt.cos(phi_ang), p2*mt.sin(phi_ang), p2*mt.cos(phi_ang), 0, p5, 0, p4, p1*mt.sin(phi_ang), p1*mt.cos(phi_ang)])
    return rhs_vector

## Assessment

Define Python variables
`D`, `H`, `THETA`, `P1`, `P2`, `P3`, `P4`, and `P5`
for storing the values of parameters
$d$, $h$, $\theta$, $p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
respectively.

Every time you solve for $\boldsymbol{\phi}$,
- use `get_coeffs_rhs()` to generate the coefficient matrix and the right-hand-side vector,
  respectively storing them in Python variables `K` and `d`;
- show the determinant and the condition number of the coefficient matrix;
  and
- print $\boldsymbol{\phi}$.


### Task 1

Determine the member forces when $C$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_1` to store $\boldsymbol{\phi}$ for this task.

In [3]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

#forces in kilonewtons
p1 = 1 #load on joint C
p2 = 0 
p3 = 0
p4 = 0
p5 = 0

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_1 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_1[0],"kN")
print("Force Member BC is:", phi_1[1],"kN")
print("Force Member BD is:", phi_1[2],"kN")
print("Force Member CD is:", phi_1[3],"kN")
print("Force Member CE is:", phi_1[4],"kN")
print("Force Member DE is:", phi_1[5],"kN")
print("Force Member DF is:", phi_1[6],"kN")
print("Force Member EF is:", phi_1[7],"kN")
print("Force Member EG is:", phi_1[8],"kN")
print("Force Member FG is:", phi_1[9],"kN")

Force Member AC is: -1.166666666666667 kN
Force Member BC is: -1.536590742882148 kN
Force Member BD is: -0.0 kN
Force Member CD is: 0.0 kN
Force Member CE is: 0.0 kN
Force Member DE is: -0.0 kN
Force Member DF is: -0.0 kN
Force Member EF is: -0.0 kN
Force Member EG is: 0.0 kN
Force Member FG is: -0.0 kN


### Task 2

Determine the member forces when $E$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_2` to store $\boldsymbol{\phi}$ for this task.

In [4]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

#forces in kilonewtons
p1 = 0 
p2 = 1 #load on joint E
p3 = 0
p4 = 0
p5 = 0

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_2 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_2[0],"kN")
print("Force Member BC is:", phi_2[1],"kN")
print("Force Member BD is:", phi_2[2],"kN")
print("Force Member CD is:", phi_2[3],"kN")
print("Force Member CE is:", phi_2[4],"kN")
print("Force Member DE is:", phi_2[5],"kN")
print("Force Member DF is:", phi_2[6],"kN")
print("Force Member EF is:", phi_2[7],"kN")
print("Force Member EG is:", phi_2[8],"kN")
print("Force Member FG is:", phi_2[9],"kN")

Force Member AC is: 1.1666666666666674 kN
Force Member BC is: -0.7682953714410738 kN
Force Member BD is: 0.5590169943749476 kN
Force Member CD is: -1.1180339887498951 kN
Force Member CE is: -0.7500000000000003 kN
Force Member DE is: -1.2500000000000002 kN
Force Member DF is: -0.0 kN
Force Member EF is: -0.0 kN
Force Member EG is: 0.0 kN
Force Member FG is: -0.0 kN


### Task 3

Determine the member forces when $G$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_3` to store $\boldsymbol{\phi}$ for this task.

In [5]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

#forces in kilonewtons
p1 = 0 
p2 = 0 
p3 = 1 #load on joint G
p4 = 0
p5 = 0

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_3 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_3[0],"kN")
print("Force Member BC is:", phi_3[1],"kN")
print("Force Member BD is:", phi_3[2],"kN")
print("Force Member CD is:", phi_3[3],"kN")
print("Force Member CE is:", phi_3[4],"kN")
print("Force Member DE is:", phi_3[5],"kN")
print("Force Member DF is:", phi_3[6],"kN")
print("Force Member EF is:", phi_3[7],"kN")
print("Force Member EG is:", phi_3[8],"kN")
print("Force Member FG is:", phi_3[9],"kN")

Force Member AC is: 0.49999999999999983 kN
Force Member BC is: 1.907319499384266e-17 kN
Force Member BD is: -1.1180339887498947 kN
Force Member CD is: -2.7755575615628907e-17 kN
Force Member CE is: -0.49999999999999983 kN
Force Member DE is: -3.103167691559091e-17 kN
Force Member DF is: 1.1180339887498947 kN
Force Member EF is: -0.0 kN
Force Member EG is: 0.49999999999999983 kN
Force Member FG is: -1.1180339887498947 kN


### Task 4

Determine the member forces when $F$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_4` to store $\boldsymbol{\phi}$ for this task.

In [6]:
d = 1
h = 6
phi_ang = mt.atan(h/3*d)
theta = 21

#forces in kilonewtons
p1 = 0 
p2 = 0 
p3 = 0 
p4 = 1
p5 = 0

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_4 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_4[0],"kN")
print("Force Member BC is:", phi_4[1],"kN")
print("Force Member BD is:", phi_4[2],"kN")
print("Force Member CD is:", phi_4[3],"kN")
print("Force Member CE is:", phi_4[4],"kN")
print("Force Member DE is:", phi_4[5],"kN")
print("Force Member DF is:", phi_4[6],"kN")
print("Force Member EF is:", phi_4[7],"kN")
print("Force Member EG is:", phi_4[8],"kN")
print("Force Member FG is:", phi_4[9],"kN")

Force Member AC is: -0.37267799624996506 kN
Force Member BC is: 0.6871842709362765 kN
Force Member BD is: -0.0 kN
Force Member CD is: 1.0 kN
Force Member CE is: 0.0 kN
Force Member DE is: -0.0 kN
Force Member DF is: -0.0 kN
Force Member EF is: -0.0 kN
Force Member EG is: 0.0 kN
Force Member FG is: -0.0 kN


### Task 5

Determine the member forces when $D$ is loaded by a unit force and all other joints are unloaded.

Define a Python variable `phi_5` to store $\boldsymbol{\phi}$ for this task.

In [7]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

#forces in kilonewtons
p1 = 0 
p2 = 0 
p3 = 0 
p4 = 0
p5 = 1

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_5 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_5[0],"kN")
print("Force Member BC is:", phi_5[1],"kN")
print("Force Member BD is:", phi_5[2],"kN")
print("Force Member CD is:", phi_5[3],"kN")
print("Force Member CE is:", phi_5[4],"kN")
print("Force Member DE is:", phi_5[5],"kN")
print("Force Member DF is:", phi_5[6],"kN")
print("Force Member EF is:", phi_5[7],"kN")
print("Force Member EG is:", phi_5[8],"kN")
print("Force Member FG is:", phi_5[9],"kN")

Force Member AC is: 0.37267799624996506 kN
Force Member BC is: 0.34359213546813827 kN
Force Member BD is: -0.24999999999999994 kN
Force Member CD is: 0.4999999999999999 kN
Force Member CE is: -0.5590169943749475 kN
Force Member DE is: 0.5590169943749473 kN
Force Member DF is: -0.0 kN
Force Member EF is: -1.0 kN
Force Member EG is: 0.0 kN
Force Member FG is: -0.0 kN


### Task 6

Determine the member forces
when $C$, $E$, $G$, $F$, and $D$
are loaded by unit forces.

Define a Python variable `phi_6` to store $\boldsymbol{\phi}$ for this task.

In [8]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 21

#forces in kilonewtons
p1 = 1
p2 = 1 
p3 = 1 
p4 = 1
p5 = 1

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_6 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_6[0],"kN")
print("Force Member BC is:", phi_6[1],"kN")
print("Force Member BD is:", phi_6[2],"kN")
print("Force Member CD is:", phi_6[3],"kN")
print("Force Member CE is:", phi_6[4],"kN")
print("Force Member DE is:", phi_6[5],"kN")
print("Force Member DF is:", phi_6[6],"kN")
print("Force Member EF is:", phi_6[7],"kN")
print("Force Member EG is:", phi_6[8],"kN")
print("Force Member FG is:", phi_6[9],"kN")

Force Member AC is: 0.49999999999999967 kN
Force Member BC is: -1.2741097079188068 kN
Force Member BD is: -0.8090169943749472 kN
Force Member CD is: 0.381966011250105 kN
Force Member CE is: -1.8090169943749472 kN
Force Member DE is: -0.6909830056250528 kN
Force Member DF is: 1.1180339887498947 kN
Force Member EF is: -1.0 kN
Force Member EG is: 0.49999999999999983 kN
Force Member FG is: -1.1180339887498947 kN


### Task 7

Similar to Task 6,
determine the member forces
when $C$, $E$, $G$, $F$, and $D$
are loaded by unit forces,
but use the results in Tasks 1 - 5.

Define a Python variable `phi_7` to store $\boldsymbol{\phi}$ for this task.

In [9]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_7 = phi_1 + phi_2 + phi_3 + phi_4 + phi_5
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_7[0],"kN")
print("Force Member BC is:", phi_7[1],"kN")
print("Force Member BD is:", phi_7[2],"kN")
print("Force Member CD is:", phi_7[3],"kN")
print("Force Member CE is:", phi_7[4],"kN")
print("Force Member DE is:", phi_7[5],"kN")
print("Force Member DF is:", phi_7[6],"kN")
print("Force Member EF is:", phi_7[7],"kN")
print("Force Member EG is:", phi_7[8],"kN")
print("Force Member FG is:", phi_7[9],"kN")

Force Member AC is: 0.5000000000000002 kN
Force Member BC is: -1.2741097079188068 kN
Force Member BD is: -0.809016994374947 kN
Force Member CD is: 0.38196601125010476 kN
Force Member CE is: -1.8090169943749477 kN
Force Member DE is: -0.6909830056250529 kN
Force Member DF is: 1.1180339887498947 kN
Force Member EF is: -1.0 kN
Force Member EG is: 0.49999999999999983 kN
Force Member FG is: -1.1180339887498947 kN


### Task 8

Determine the member forces using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$.

Define a Python variable `phi_8` to store $\boldsymbol{\phi}$ for this task.

In [10]:
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

#forces in kilonewtons
p1 = 1
p2 = 9
p3 = 8
p4 = 7
p5 = 5

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_8 = np.linalg.solve(K, d)
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_8[0],"kN")
print("Force Member BC is:", phi_8[1],"kN")
print("Force Member BD is:", phi_8[2],"kN")
print("Force Member CD is:", phi_8[3],"kN")
print("Force Member CE is:", phi_8[4],"kN")
print("Force Member DE is:", phi_8[5],"kN")
print("Force Member DF is:", phi_8[6],"kN")
print("Force Member EF is:", phi_8[7],"kN")
print("Force Member EG is:", phi_8[8],"kN")
print("Force Member FG is:", phi_8[9],"kN")

Force Member AC is: 12.587977340833408 kN
Force Member BC is: -1.9229985119571857 kN
Force Member BD is: -5.163118960624629 kN
Force Member CD is: -0.5623058987490573 kN
Force Member CE is: -13.54508497187474 kN
Force Member DE is: -8.454915028125267 kN
Force Member DF is: 8.944271909999157 kN
Force Member EF is: -5.0 kN
Force Member EG is: 3.9999999999999987 kN
Force Member FG is: -8.944271909999157 kN


### Task 9

Similar to Task 8,
determine the member forces using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
but use the results in Tasks 1 - 5.


Define a Python variable `phi_9` to store $\boldsymbol{\phi}$ for this task.

In [11]:
#using set 3 values
#converting feet to meters
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_9 = phi_1 + 9*phi_2 + 8*phi_3 + 7*phi_4 + 5*phi_5
#print separately the values of the unknown variables:
print("Force Member AC is:", phi_9[0],"kN")
print("Force Member BC is:", phi_9[1],"kN")
print("Force Member BD is:", phi_9[2],"kN")
print("Force Member CD is:", phi_9[3],"kN")
print("Force Member CE is:", phi_9[4],"kN")
print("Force Member DE is:", phi_9[5],"kN")
print("Force Member DF is:", phi_9[6],"kN")
print("Force Member EF is:", phi_9[7],"kN")
print("Force Member EG is:", phi_9[8],"kN")
print("Force Member FG is:", phi_9[9],"kN")

Force Member AC is: 12.587977340833408 kN
Force Member BC is: -1.9229985119571844 kN
Force Member BD is: -5.1631189606246295 kN
Force Member CD is: -0.5623058987490563 kN
Force Member CE is: -13.54508497187474 kN
Force Member DE is: -8.454915028125265 kN
Force Member DF is: 8.944271909999157 kN
Force Member EF is: -5.0 kN
Force Member EG is: 3.9999999999999987 kN
Force Member FG is: -8.944271909999157 kN


In [12]:
#Just to verify if equations are at equilibrium
#using set 3 values
#converting feet to meters
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

p1 = 1
p2 = 1 
p3 = 1 
p4 = 1
p5 = 1

K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)
d = delta(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

phi_test = np.linalg.solve(K, d)

print((K @ phi_test) - d)#must be equal to 0 or just infinitely close to 0

[-1.11022302e-16  0.00000000e+00 -1.11022302e-16  0.00000000e+00
  0.00000000e+00  0.00000000e+00  5.55111512e-17  0.00000000e+00
 -2.22044605e-16  5.55111512e-17]


In [13]:
#checks determinant
d=1
h=6
phi_ang = mt.atan(h/3*d)
theta = 15

p1 = 1
p2 = 9 
p3 = 8 
p4 = 7
p5 = 5
K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

np.linalg.cond(K)

np.float64(7.113287080521642)

In [14]:
K = get_coeffs_rhs(d, h, theta, p1, p2, p3, p4, p5, phi_ang)

np.linalg.det(K)

np.float64(-0.46566840006975924)

### Task 10

Determine the magnitudes of the support reactions
using the assigned values of
$p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$.

Define Python variables `rxn_A` and `rxn_B` 
to store the forces developed at $A$ and $B$, respectively.

In [15]:
rxn_A_x = -phi_8[0]*mt.cos(phi_ang)
rxn_A_y = -phi_8[0]*mt.sin(phi_ang)
phi_ang = mt.atan(h/3*d)
theta = 15

print(f"Ax is: {rxn_A_x}")
print(f"Ay is: {rxn_A_y}")
print(f"A is: {mt.sqrt(rxn_A_x**2+rxn_A_y**2)}") #in kN

Ax is: -5.629514606666106
Ay is: -11.259029213332218
A is: 12.58797734083341


In [16]:
# For reactions B and its components
phi_ang = mt.atan(h/3*d)
theta = 15
BC_x_vec = d/(mt.sqrt(d**2 + (2*h/3)**2))
BC_y_vec = 2*h/(3*mt.sqrt(d**2 + (2*h/3)**2))

bx = -phi_8[3] - phi_8[1]*BC_x_vec
by = phi_8[1]*BC_y_vec
    
print(f"Bx is:, {bx}")
print(f"By is:, {by}")
print(f"B is: {mt.sqrt(bx**2+by**2)}") #in kN

Bx is:, 1.0287015447905317
By is:, -1.8655825841658973
B is: 2.1304049489703907


## Instructions

Do not use any library or module other than those in the Imports section.

### Parameters

For each set,
the values shown are that of
$d$, $h$, $\theta$, $p_{1}$, $p_{2}$, $p_{3}$, $p_{4}$, and $p_{5}$,
respectively.

- Set 1: 1.5 ft, 4.5 ft, 21°, 5 kN, 7 kN, 6 kN, 8 kN, 1 kN
- Set 2: 2.0 ft, 3.0 ft, 17°, 9 kN, 6 kN, 5 kN, 1 kN, 7 kN
- Set 3: 1.0 ft, 6.0 ft, 15°, 1 kN, 9 kN, 8 kN, 7 kN, 5 kN

### Scoring

In each of Tasks 1 - 9,
obtaining a correct member force merits one (1) point.

In each Task but 7, 9, and 10,
- obtaining a correct determinant merits one (1) point,
  and
- obtaining a correct condition number merits one (1) point.

For Task 10,
obtaining a correct reaction force merits five (5) points.

Every violation of an instruction
means a deduction of one (1) point.

All in all, one may earn up to 124 points for this activity.

### Submission

Download this notebook file,
and save with a filename following the pattern
`ACT-08_<Group name>`,
where the group name is as listed in class.
For example, if you belong to the group SixIsEven,
then your notebook should be named `ACT-08_SixIsEven.ipynb`.
Submit your notebook via the classwork platform for this activity in Google Classroom.
Submissions beyond the deadline will not be considered.

Lastly, the use of AI tools to answer this exam is not prohibited,
but it is of ethical interest to disclose such use.
This is in line with the
[MSU Policy on the Fair and Ethical Use of AI and Its Applications](https://www.msumain.edu.ph/wp-content/uploads/2024/05/MSU-Policy-on-Ethical-use-of-AI-Policies.pdf).
As such, please include a brief statement (in a private comment to this classwork)
declaring which and how AI tools are used in your work.

*Last updated by Christian Cahig on 2025-11-25*